packages/r2/test/r2Object.spec.ts

```diff
@@ -496,3 +503,36 @@ test("R2Object: R2ObjectBody: push 'body' ReadableStream to Response", async (t)
   const testResponse = new Response(readable);
   t.true(testResponse.body instanceof ReadableStream);
 });
+
+const bufferChecksums: R2Checksums<Buffer> = {
+  md5: crypto.createHash("md5").update("test").digest(),
+  sha1: crypto.createHash("sha1").update("test").digest(),
+  sha256: crypto.createHash("sha256").update("test").digest(),
+  sha384: crypto.createHash("sha384").update("test").digest(),
+  sha512: crypto.createHash("sha512").update("test").digest(),
+};
+const stringChecksums: R2Checksums<string> = Object.fromEntries(
+  Object.entries(bufferChecksums as { [key: string]: Buffer }).map(
+    ([key, value]) => [key, value.toString("hex")]
+  )
+);
+test("Checksums: returns `ArrayBuffer` hashes", async (t) => {
+  const checksums = new Checksums(stringChecksums);
+  t.deepEqual(checksums.md5, viewToBuffer(bufferChecksums.md5!));
+  t.deepEqual(checksums.sha1, viewToBuffer(bufferChecksums.sha1!));
+  t.deepEqual(checksums.sha256, viewToBuffer(bufferChecksums.sha256!));
+  t.deepEqual(checksums.sha384, viewToBuffer(bufferChecksums.sha384!));
+  t.deepEqual(checksums.sha512, viewToBuffer(bufferChecksums.sha512!));
+
+  // Check getter returns new `ArrayBuffer` each time
+  const buffer = checksums.md5;
+  assert(buffer !== undefined);
+  new MessageChannel().port1.postMessage(buffer, [buffer]);
+  t.is(buffer.byteLength, 0); // (detached)
+  t.is(checksums.md5?.byteLength, 16); // New buffer
+});
+test("Checksums: returns hex hashes with JSON.stringify()", async (t) => {
+  const checksums = new Checksums(stringChecksums);
+  t.deepEqual(checksums.toJSON(), stringChecksums);
+  t.is(JSON.stringify(checksums), JSON.stringify(stringChecksums));
+});
```

Comment:
I'm so jealous of how easy it is to write miniflare tests compared to EW tests :cry: